File: SecondYear/semestrul4/MPP/PROIECT/CSharpProjectNetworking/Server/service/AbstractService.cs
using Model;
using Persistence;

namespace Server;
public class AbstractService<ID, E> : IService<ID, E> where E : Entity<ID>
{
    protected IRepository<ID, E> repository;

    public AbstractService(IRepository<ID, E> repository)
    {
        this.repository = repository;
    }

    public virtual E FindOne(ID id)
    {
        return repository.FindOne(id);
    }

    public virtual IEnumerable<E> FindAll()
    {
        return repository.FindAll();
    }

    public virtual void Save(E entity)
    {
        repository.Save(entity);
    }
}

File: SecondYear/semestrul4/MPP/examen_practic/Rezolvari/SUB3_2024/server/src/main/java/org/server/service/AbstractService.java
package org.server.service;

import model.Entity;
import persistence.Repository;

import java.io.Serializable;
import java.sql.SQLException;
import java.util.Optional;

public class AbstractService<ID extends Serializable, E extends Entity<ID>> implements Service<ID, E>{
    Repository<ID, E> repository;
    public AbstractService(Repository<ID, E> repository) {
        this.repository = repository;
    }

    @Override
    public Optional<E> findOne(ID id) throws SQLException {
        return repository.findOne(id);
    }

    @Override
    public Iterable<E> findAll() {
        return repository.findAll();
    }

    @Override
    public void save(E entity) throws SQLException {
        repository.save(entity);
    }

    @Override
    public void delete(ID id) throws SQLException {

    }

    @Override
    public Optional<E> update(E entity) throws SQLException {
        return Optional.empty();
    }

}
